packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -561,6 +561,8 @@ export abstract class BaseService extends Resource
       ...additionalProps,
     });
 
+    this.node.addDependency(this.taskDefinition.taskRole);
```

Comment:
From the linked issue it looks like we should be adding a dependency on the `Policy` not the role. There should already be an implicit dependency on the role and in cases where the policy is updated the role itself will probably not be updated so will not be in the dependency chain.

Reply:
Thanks for the feedback.

The [default policy](https://github.com/aws/aws-cdk/blob/3fe166f8f74784eb04b6e048659e46785483d786/packages/aws-cdk-lib/aws-iam/lib/role.ts#L413) is a `private` property in the role.
Should I create a public getter to retrieve it and add a dependency with:
```
this.node.addDependency(this.taskDefinition.taskRole.getDefaultPolicy);
```
?

Reply:
I don't think dependencies work that way. If I declare an explicit dependency on resource A, CloudFormation does not view that as adding an explicit dependency on every resource that resource A depends on. It only depends on resource A, so if resource A is not updated then it doesn't have to wait.

But we can always run tests to confirm :)

Reply:
We might be able to accomplish this globally though if we can assume that anytime someone adds a dependency on a role that they also want to add a dependency on the default policy (I think that is a safe assumption)

I think by adding something like this to `Role`
```ts
Dependable.implement(this, {
  dependencyRoots: [this, this.defaultPolicy],
});
```

Whenever you do `addDependency(role)` it will also add a dependsOn for the policy.

Reply:
> I don't think dependencies work that way. If I declare an explicit dependency on resource A, CloudFormation does not view that as adding an explicit dependency on every resource that resource A depends on. It only depends on resource A, so if resource A is not updated then it doesn't have to wait.

For sure, but I'm only talking about child constructs of the Role. The default policy is scoped to the role, so addDependency(role) takes care of it.

Reply:
ah you are right. @lpizzinidev can you add a test that asserts the dependency is added on the policy as well?